src/overlays/actors/ovl_En_Hg/z_en_hg.c

```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
+    Player* player = PLAYER;
+    s32 pad;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/EnHg_Destroy.asm")
+    this->actor.speedXZ = 1.6f;
+    if (!(player->stateFlags2 & 0x08000000)) {
+        if (func_80152498(&globalCtx->msgCtx) == 0) {
+            if (((this->skelAnime.animCurrentFrame > 9.0f) && (this->skelAnime.animCurrentFrame < 16.0f)) ||
+                (this->skelAnime.animCurrentFrame > 44.0f) && (this->skelAnime.animCurrentFrame < 51.0f)) {
+                Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+                Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x3E8, 0x14);
+                this->actor.world.rot.y = this->actor.shape.rot.y;
+            }
+            if ((Math_Vec3f_DistXZ(&this->actor.world.pos, &this->actor.home.pos) > 200.0f) &&
+                (Math_Vec3f_DistXZ(&player->actor.world.pos, &this->actor.home.pos) > 200.0f)) {
+                func_80BCF5F0(this);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF354.asm")
+void func_80BCF5F0(EnHg* this) {
+    func_800BDC5C(&this->skelAnime, animations, 0);
+    this->actionFunc = func_80BCF634;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF398.asm")
+void func_80BCF634(EnHg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF468.asm")
+    if (Math_Vec3f_DistXZ(&player->actor.world.pos, &this->actor.home.pos) < 200.0f) {
+        func_80BCF468(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF4AC.asm")
+void func_80BCF68C(EnHg* this) {
+    func_800BDC5C(&this->skelAnime, animations, 2);
+    this->actionFunc = func_80BCF6D0;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF5F0.asm")
+void func_80BCF6D0(EnHg* this, GlobalContext* globalCtx) {
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
+        func_80BCF5F0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF634.asm")
+void func_80BCF710(EnHg* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 0) {
+        if (func_800B84D0(&this->actor, globalCtx)) {
+            func_801518B0(globalCtx, 0x24FU, &this->actor);
+        } else {
+            func_800B8614(&this->actor, globalCtx, 80.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF68C.asm")
+void func_80BCF778(EnHg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health == 1) {
+        if (this->actionFunc == func_80BCF4AC || this->actionFunc == func_80BCF6D0 ||
+            this->actionFunc == func_80BCF634) {
+            func_800B9010(&this->actor, 0x32B7);
```

Comment:
```suggestion
            func_800B9010(&this->actor, NA_SE_EN_HALF_REDEAD_LOOP - SFX_FLAG);
```
also an audio function